Sources/Passes/Passes.docc/GettingStarted.md

```diff
@@ -0,0 +1,355 @@
+            self.auxiliaryFields = [.init(key: "auxiliary", label: "Auxiliary", value: "Auxiliary")]
+            self.backFields = [.init(key: "back", label: "Back", value: "Back")]
+            self.transitType = transitType
+        }
+    }
+
+    init(data: PassData, pass: PKPass) {
+        self.description = data.title
+        self.serialNumber = pass.id!.uuidString
+        self.authenticationToken = pass.authenticationToken
+    }
+}
+```
+
+> Important: You **must** add `api/passes/` to your `webServiceURL`, as shown in the example above.
+
+### Implement the Delegate
+
+Create a delegate class that implements ``PassesDelegate``.
+In the ``PassesDelegate/sslSigningFilesDirectory`` you specify there must be the `WWDR.pem`, `passcertificate.pem` and `passkey.pem` files.
+If they are named like that you're good to go, otherwise you have to specify the custom name.
+
+> Tip: Obtaining the three certificates files could be a bit tricky. You could get some guidance from [this guide](https://github.com/alexandercerutti/passkit-generator/wiki/Generating-Certificates) and [this video](https://www.youtube.com/watch?v=rJZdPoXHtzI).
+
+There are other fields available which have reasonable default values. See ``PassesDelegate``'s documentation.
+
+Because the files for your pass' template and the method of encoding might vary by pass type, you'll be provided the ``PKPass`` for those methods.
+In the ``PassesDelegate/encode(pass:db:encoder:)`` method, you'll want to encode a `struct` that conforms to ``PassJSON``.
+
+```swift
+import Vapor
+import Fluent
+import Passes
+
+final class PassDelegate: PassesDelegate {
+    let sslSigningFilesDirectory = URL(fileURLWithPath: "Certificates/Passes/", isDirectory: true)
+
+    let pemPrivateKeyPassword: String? = Environment.get("PEM_PRIVATE_KEY_PASSWORD")!
+
+    func encode<P: PassModel>(pass: P, db: Database, encoder: JSONEncoder) async throws -> Data {
+        // The specific PassData class you use here may vary based on the `pass.passTypeIdentifier`
+        // if you have multiple different types of passes, and thus multiple types of pass data.
+        guard let passData = try await PassData.query(on: db)
+            .filter(\.$pass.$id == pass.requireID())
+            .first()
+        else {
+            throw Abort(.internalServerError)
+        }
+        guard let data = try? encoder.encode(PassJSONData(data: passData, pass: pass)) else {
+            throw Abort(.internalServerError)
+        }
+        return data
+    }
+
+    func template<P: PassModel>(for pass: P, db: Database) async throws -> URL {
+        // The location might vary depending on the type of pass.
+        return URL(fileURLWithPath: "Templates/Passes/", isDirectory: true)
+    }
+}
+```
+
+> Important: You **must** explicitly declare ``PassesDelegate/pemPrivateKeyPassword`` as a `String?` or Swift will ignore it as it'll think it's a `String` instead.
```

Comment:
Why do we need this as a `String?` in particular?

Reply:
Because it isn't required for the pem private key to be encrypted, so in PassesServiceCustom it unwraps it and acts accordingly if it exists or if it's nil